src/printer.rs

```diff
@@ -75,85 +93,173 @@ impl<'a> Printer<'a> {
-            self.handle,
-            "{}",
-            decorations
-                .into_iter()
-                .filter_map(|dec| if grid_requested {
-                    Some(dec.unwrap_or_else(|| " ".to_owned()))
-                } else {
-                    dec
-                })
-                .collect::<Vec<_>>()
-                .join(" ")
-        )?;
+        let mut cursor: usize = 0;
+        let mut cursor_max: usize = self.config.term_width;
+
+        // Line decoration.
+        let decorations = self.gen_decorations(line_number);
+        let gutter_width = decorations.len() + decorations.iter().fold(0, |a, x| a + x.size);
+
+        if gutter_width > 0 {
+            cursor_max -= gutter_width;
+            write!(
+                self.handle,
+                "{} ",
+                decorations
+                    .iter()
+                    .map(|seg| seg.text.to_owned())
+                    .collect::<Vec<String>>()
+                    .join(" ")
+            )?;
+        }
+
+        // Grid border.
+        let border = if gutter_width > 0 && self.config.output_components.grid() {
+            self.gen_border()
+        } else {
+            PrintSegment {
+                size: 0,
+                text: "".to_owned(),
+            }
+        };
+
+        cursor_max -= border.size;
+        write!(self.handle, "{}", border.text)?;
+
+        // Line contents.
+        if self.config.output_wrap == OutputWrap::None {
+            let true_color = self.config.true_color;
+            let colored_output = self.config.colored_output;
+
+            write!(self.handle, "{}",
+                   regions.iter()
+                       .map(|&(style, text)| as_terminal_escaped(style, text, true_color, colored_output))
+                       .collect::<Vec<_>>()
+                       .join("")
+            )?;
+        } else {
+            for &(style, text) in regions.iter() {
+                let mut chars = text.chars().filter(|c| *c != '\n');
+                let mut remaining = text.chars().filter(|c| *c != '\n').count();
+
+                while remaining > 0 {
+                    let available = cursor_max - cursor;
+
+                    // It fits.
+                    if remaining <= available {
+                        let text = chars.by_ref().take(remaining).collect::<String>();
+                        cursor += remaining;
+
+                        write!(
+                            self.handle,
+                            "{}",
+                            as_terminal_escaped(
+                                style,
+                                &*text,
+                                self.config.true_color,
+                                self.config.colored_output,
+                            )
+                        )?;
+                        break;
+                    }
+
+                    // It wraps.
+                    if self.config.output_wrap == OutputWrap::Character {
```

Comment:
this is the `else` part, so we already know that this will evaluate to true every time. or am I missing something?

Reply:
Yeah, it evaluates to true every time. I left it in there in case different wrapping styles were added in the future, but on second thought, it's unnecessary right now.

Reply:
I did that it there in case different wrapping styles would be added in the future, but looking back at it, it's definitely not necessary right now. I'll remove it.